source/s3_auto_ranged_get.c

```diff
@@ -739,17 +791,32 @@ static void s_s3_auto_ranged_get_request_finished(
 
             copy_http_headers(request->send_data.response_headers, response_headers);
 
-            /* If this request is a part, then the content range isn't applicable. */
             if (request->request_tag == AWS_S3_AUTO_RANGE_GET_REQUEST_TYPE_GET_OBJECT_WITH_RANGE ||
                 request->request_tag == AWS_S3_AUTO_RANGE_GET_REQUEST_TYPE_GET_OBJECT_WITH_PART_NUMBER_1) {
-                /* For now, we can assume that discovery of size via the first part of the object does not apply to
-                 * breaking up a ranged request. If it ever does, then we will need to repopulate this header. */
-                AWS_ASSERT(!auto_ranged_get->initial_message_has_range_header);
-                aws_http_headers_erase(response_headers, g_content_range_header_name);
+
+                if (!auto_ranged_get->initial_message_has_range_header) {
+                    /* content range isn't applicable. */
+                    aws_http_headers_erase(response_headers, g_content_range_header_name);
+                } else {
+                    /* Populate the header with object_range */
+                    char content_range_buffer[64] = "";
+                    snprintf(
+                        content_range_buffer,
+                        sizeof(content_range_buffer),
+                        "bytes %" PRIu64 "-%" PRIu64 "/%" PRIu64,
+                        object_range_start,
+                        object_range_end,
+                        object_size);
+                    aws_http_headers_set(
+                        response_headers,
+                        g_content_range_header_name,
+                        aws_byte_cursor_from_c_str(content_range_buffer));
+                }
             }
 
+            uint64_t content_length = object_size ? object_range_end - object_range_start + 1 : 0;
```

Comment:
what happens if `object_size` is 0 and `initial_message_has_range_header`? Does that always result in `empty_file_error`?

I just want to be sure the we're fabricating the correct Content-Range header a few lines up

Reply:
It will result in 416 Range not satisfiable error, and we will never trigger the onHeaders as we will fail to find the object_size at https://github.com/awslabs/aws-c-s3/blob/main/source/s3_auto_ranged_get.c#L612.